### PR TITLE
feat(argo-workflows): support resourceRateLimit config

### PR DESCRIPTION
**Pull request by clementlecorre**
Hi,

I would like add this feature `resourceRateLimit`. https://github.com/argoproj/argo-workflows/blob/a29495bb5b4aedcbd8fa7bc433c1bccbc374e6d1/docs/workflow-controller-configmap.yaml#L26-L31

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
Please rebase and bump the minor version of the chart.
In the meantime we introduced helm-docs for the workflows chart. So please use something like:

```yaml
  # -- Globally limits the rate at which pods are created.
  # This is intended to mitigate flooding of the Kubernetes API server by workflows with a large amount of
  # parallel nodes.
  resourceRateLimit: {}
    # limit: 10
    # burst: 1
```

File: charts/argo-workflows/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: argo-workflows
 description: A Helm chart for Argo Workflows
 type: application
-version: 0.8.3
+version: 0.8.4
```
Comment (mkilchhofer) [inline charts/argo-workflows/Chart.yaml:5]:
```suggestion
version: 0.9.0
```

Comment (clementlecorre) [reply charts/argo-workflows/Chart.yaml]: Done! ✅

**Review by mkilchhofer — APPROVED** [review]
lgtm

**Pull request merged**